### ISSUE TITLE
Pod still running after deployment deleted

### ISSUE THREAD

**Issue by kavaribes** [godaddy/kubernetes-client#536]
I'm using those methods for creating and removing k8s deployment, service, horizontalpodautoscaler, poddisruptionbudget. 

`client.apis.extensions.v1beta1.namespaces(namespaceName).deployments.post()`
`client.apis.extensions.v1beta1.namespaces(namespaceName).deployments(deploymentName).delete()`

`client.api.v1.namespaces(namespaceName).services.post()`
`client.api.v1.namespaces(namespaceName).services(deploymentName).delete()`

`client.apis.autoscaling.v1.namespaces(namespaceName).horizontalpodautoscalers.post()`
`client.apis.autoscaling.v1.namespaces(namespaceName).horizontalpodautoscalers(deploymentName).delete()`

`client.apis.policy.v1beta1.namespaces(namespaceName).poddisruptionbudgets.post()`
`client.apis.policy.v1beta1.namespaces(namespaceName).poddisruptionbudgets(deploymentName).delete()`

And creation step works fine. After step of removing the deployment, service, horizontalpodautoscaler, poddisruptionbudget removing correct, but pod still running, and I can't remove it. I need create deployment again through `kubectl` and remove deployment through it as well. Moreover when the pod is "hanging" arise an issue with migrating of nodes inside Kubernetes.

Vesion of module 8.3.3

**Comment by silasbw** [comment]
Duplicate of https://github.com/godaddy/kubernetes-client/issues/148

tl;dr this is the expected behavior of the kubernetes API
